src/tir/ir/expr.cc

```diff
@@ -1059,11 +1059,13 @@ TVM_STATIC_IR_FUNCTOR(ReprPrinter, vtable)
 
 // BufferLoad
 void BufferLoadNode::LegalizeDType() {
-  int index_lanes = 1;
-  for (const auto& index : indices) {
-    index_lanes *= index.dtype().lanes();
+  for (size_t i = 0; i < indices.size(); i++) {
```

Comment:
You can just ignore the last index (i.e. `i < indices.size()-1`), then you don't need the `i + 1 == end` check.  Also, you can use `indices[i].dtype.is_scalar()` instead of checking lanes directly.

Reply:
I like these, and it is definitely cleaner with the `is_scalar()` helper.  Changes made as suggested.